Kernel/KBO.hpp

```diff
@@ -186,6 +185,83 @@ class KBO
   template<class SigTraits> 
   void showConcrete_(std::ostream&) const;
 
+  /**
+   * Class to represent the current state of the KBO comparison.
+   * Based on Bernd Loechner's "Things to Know when Implementing KBO"
+   * (https://doi.org/10.1007/s10817-006-9031-4)
+   * @since 30/04/2008 flight Brussels-Tel Aviv
+   */
+  class State
+  {
+  public:
+    /** Initialise the state */
+    State(KBO* kbo)
+      : _kbo(*kbo)
+    {}
```

Comment:
Would it make sense to call `init` from the constructor (or equivalent code)? As it is this innocent-looking aggregate may be uninitialised if you don't call `init` explicitly. But I don't have a strong view.

Reply:
Otherwise some parts could remain non-zero?

Reply:
Right. But might be worth it depending on the usage pattern.